libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/model/Command.kt

```diff
@@ -0,0 +1,16 @@
+package com.mapbox.navigation.dropin.model
+
+/**
+ * Actions dispatched as a response to a user direct interaction with the View.
+ * et. button click, menu item selection, checkbox toggle etc.
+ */
+internal sealed class Command : Action {
+    object RecenterCamera : Command()
+    object ToggleCameraMode : Command()
+
+    object ShowRoutePreview : Command()
+    object StartActiveNavigation : Command()
+    object EndNavigation : Command()
+
+    object ToggleAudioGuidance : Command()
+}
```

Comment:
Is this extra layer necessary?

Seems like they're duplicating existing actions 🤔 .

Reply:
It is necessary so we differentiate user-invoked actions from the system ones.
eg. If we would just intercept `CameraAction.ToFollowing` we wouldn't know if it was dispatched as a result of the user pressing the recenter button or as a result of some internal business logic.

Reply:
> Have we added support for something?

Yes. I've added `NavigationViewListener` in this PR that allows on-button click interception as specified in the original ticket.

_"Developers should be able to intercept on click events on action buttons: Recenter, Camera Control, Start Navigation, End Navigation"_


> Is it to support the ActionInterceptor?

No, it is to allow user-invoked actions detection. `ActionInterceptor` adds a new capability to the `Store` that allows interception of any action before it is passed down to reducers.

Reply:
Ok thanks for clarifying! Looks like there are other conversation threads on this topic, resolving this one.